framework/src/loops/ComputeJacobianThread.C

```diff
@@ -336,6 +336,14 @@ ComputeJacobianThread::postElement(const Elem * /*elem*/)
     Threads::spin_mutex::scoped_lock lock(Threads::spin_mtx);
     _fe_problem.addCachedJacobian(_tid);
   }
+
+#ifdef MOOSE_GLOBAL_AD_INDEXING
+  // global indexing uses a different cache Jacobian API in assembly that actually feeds values into
+  // different assembly data members. So we have to call a different addCached method in order to
+  // accumulate the values into the Jacobian
+  if (_fe_problem.haveFV())
+    _fe_problem.addCachedJacobianContributions(_tid);
```

Comment:
Is there a reason we can't push this ifdef and logic down into FEProblem::addCachedJacobian instead of having an extra function that has to be called everywhere jacobians are being calc'd from?